src/java/org/apache/cassandra/tools/nodetool/DataPaths.java

```diff
@@ -38,16 +45,46 @@ public class DataPaths extends NodeToolCmd
     @Option(title = "format", name = {"-F", "--format"}, description = "Output format (json, yaml)")
     private String outputFormat = "";
 
+    @Option(title = "",
+            name = {"--list-orphaned"},
+            description = "List directories in the data paths which are not controlled by Cassandra")
+    private Boolean orphaned = false;
+
     @Override
     protected void execute(NodeProbe probe)
     {
         if (!outputFormat.isEmpty() && !"json".equals(outputFormat) && !"yaml".equals(outputFormat))
-        {
             throw new IllegalArgumentException("arguments for -F are yaml and json only.");
-        }
 
         DataPathsHolder holder = new DataPathsHolder(probe, tableNames);
-        StatsPrinter<DataPathsHolder> printer = DataPathsPrinter.from(outputFormat);
-        printer.print(holder, probe.output().out);
+        if (orphaned)
+        {
+            if (!outputFormat.isEmpty() || !tableNames.isEmpty())
```

Comment:
I would move this up, outside from this `if`. Because here we might fail but you were calling Cassandra by `DataPathsHolder` already and unnecessarilly. You queried Cassandra to return you stuff but you failed. I would fail as soon as possible, doing JMX query only if we are sure we are going to need it.